### PR TITLE
PM2 compatibility

### PR DESCRIPTION
**Pull request by webphax**
Check on startup if `process.stdin.setRawMode` is available and load `setupKeyInput()` if so. Otherwise skip loading `setupKeyInput()`.  This provides compatibility with process managers such as PM2.

### CODE REVIEW

**Comment by samsonmking** [comment]
@webphax This looks good to me. It looks like this branch pulls in some of the grayscale commits, too. Let's either wait until that PR is merged or cherry pick just the PM2 commit into a separate PR.

**Comment by webphax** [comment]
Yes, this is fully based (but not depend) on the 1Bit4Grey branch as I first thought it migth have a dependency. The only relevant commit for this is 4be0594.

**Comment by samsonmking** [comment]
@webphax can you please cherry pick 4be0594 and force push just that commit to this pr?

**Comment by webphax** [comment]
> @webphax can you please cherry pick [4be0594](https://github.com/samsonmking/epaper.js/commit/4be0594ed17d5eea4ad23ff7eb718a33a745b089) and force push just that commit to this pr?

done :)

**Comment by samsonmking** [comment]
@webphax I was actually thinking about this a little more. PM2 is pretty feature rich, but some of the features seem like they wouldn't really be relevant to this project, like load balancing.

I think supporting the ability to run epaper.js as a daemon is a good idea! However, I'd like to keep it as light weight as possible, to not take up too many resources on lower power devices, like the Raspberry Pi Zero.

Systemd, built into Rasbian might the simplest solution. It also supports other features we might want, like automatically starting a process on boot.

Would you be interested in adding a systemd example to the examples directory? Here's some resources to get you started:
https://www.axllent.org/docs/nodejs-service-with-systemd/
https://www.raspberrypi.org/documentation/linux/usage/systemd.md

File: index.js
Comment (samsonmking) [inline index.js:47]: It might be a bit nicer to do this check in `setupKeyInput` instead? Something like:
```javascript
function setupKeyInput(driver) {
    if(!process.stdin.setRawMode) {
      return;
    }
    readline.emitKeypressEvents(process.stdin);
    process.stdin.setRawMode(true);
    process.stdin.addListener('keypress', (key, data) => {
        if (data.ctrl && data.name === 'c') {
            driver.sleep();
            process.exit();
        }
    });
}
```

Comment (samsonmking) [inline index.js:47]: According to [waveshare FAQ](https://www.waveshare.com/wiki/4.2inch_e-Paper_Module): 'Please set the e-Paper to sleep mode or power off it directly, otherwise, the e-Paper is damaged because of working in high voltage for long time.'

`setupKeyInput:29` ensures that the display is properly put to sleep when the process is exited with control-c. I think we need to make sure we take the same precaution when using process managers.

**Review by samsonmking — CHANGES_REQUESTED** [review]
- [ ] Put device to sleep when process exits if using process manager

File: index.js
Comment (webphax) [reply index.js]: Good point. I've added it with commit c6734e8. If possible I would like to stick with pm2 as a start. I'm currently running my application with pm2 on a Raspberry Pi Zero and can confirm that it takes some additional ressources.

About the implementation I'm not quite sure if it's still needed to check explicitly for the key input within the 'setupKeyInput' function. Otherwise it could be enough to intercept the SIGINT signal (from whatever source).

**Comment by webphax** [comment]
> Systemd, built into Rasbian might the simplest solution. It also supports other features we might want, like automatically starting a process on boot.
> 
> Would you be interested in adding a systemd example to the examples directory? Here's some resources to get you started:
> https://www.axllent.org/docs/nodejs-service-with-systemd/
> https://www.raspberrypi.org/documentation/linux/usage/systemd.md
Yes, I'll have a look at that in parallel as I'm eventually planning to switch from constant power supply to battery power supply.

**Comment by samsonmking** [comment]
@webphax I apologize for the long delay - I haven't had a chance to work on this project lately. To help me speed up testing, could you provide the steps you use to run with pm2? Once I can confirm everything is working on my end, I'll go ahead and merge the PR.

**Comment by gavinr** [comment]
I have run into this issue too. 

For the replication steps, on linux:

```
sudo npm install -g pm2
pm2 start index.js
pm2 log
```

That will show you the error:
image

**Comment by samsonmking** [comment]
@gavinr thanks for the reproduction steps. I made a couple minor changes to the changes proposed in this fix and released them as part of #24. You should now be able to install version `1.3.4 ` and use it with pm2. Please don't hesitate to reach out if you have any issues.

**Comment by gavinr** [comment]
Thank you!

**Pull request closed without merging**
